2nd-gen/alerts-to-discord/functions/index.js

```diff
@@ -86,9 +94,112 @@ id: \`${id}\`
   } catch (error) {
     logger.error(
-        `Unable to post fatal Crashlytics alert ${id} to Discord`,
+        `Unable to post fatal Crashlytics alert ${id} for ${appId} to Discord`,
         error,
     );
   }
 });
+
+/**
+ * function triggered by AppDistribution that publishes a message
+ * to Discord whenever a new iOS tester device is registered.
+ */
+// [START v2AppDistributionAlertTrigger]
+exports.postnewduuidtodiscord = onNewTesterIosDevicePublished(async (event) => {
+// [END v2AppDistributionAlertTrigger]
+  // [START v2AppDistributionEventPayload]
+  // construct a helpful message to send to Discord
+  const appId = event.appId;
+  const {
+    testerDeviceIdentifier,
+    testerDeviceModelName,
+    testerEmail,
+    testerName,
+  } = event.data.payload;
+  const message = `
+📱 New iOS device registered by ${testerName} <${testerEmail}> for ${appId}
+
+UDID **${testerDeviceIdentifier}** for ${testerDeviceModelName}
+`;
+  // [END v2AppDistributionEventPayload]
+
+  try {
+    // [START v2SendNewTesterIosDeviceToDiscord]
+    const response = await postMessageToDiscord("AppDistribution Bot", message);
+    if (response.ok) {
+      logger.info(
+          `Posted iOS device registration alert for ${testerEmail} to Discord`,
+      );
+    } else {
+      throw new Error(response.error);
+    }
+    // [END v2SendNewTesterIosDeviceToDiscord]
+  } catch (error) {
+    logger.error(
+        `Unable to post iOS device registration for ${testerEmail} to Discord`,
+        error,
+    );
+  }
+});
+// [END v2Alerts]
+
+/**
+ * Function triggered by Firebase Performance that publishes
+ * a message to Discord whenever a performance threshold alert is fired.
+ */
+// [START v2PerformanceAlertTrigger]
+exports.postperformancealerttodiscord = onThresholdAlertPublished(
+    async (event) => {
+      // [END v2PerformanceAlertTrigger]
+      // [START v2PerformanceEventPayload]
+      // construct a helpful message to send to Discord
+      const appId = event.appId;
+      const {
+        eventName,
+        metricType,
+        eventType,
+        numSamples,
+        thresholdValue,
+        thresholdUnit,
+        conditionPercentile,
+        appVersion,
+        violationValue,
+        violationUnit,
+        investigateUri,
+      } = event.data.payload;
+      const message = `
+    ⚠️ Performance Alert for ${metricType} of ${eventType}: **${eventName}** ⚠️
+    
+    App id: ${appId}
+    Alert condition: ${thresholdValue} ${thresholdUnit}
+    Percentile (if applicable): ${conditionPercentile}
+    App version (if applicable): ${appVersion}
+    
+    Violation: ${violationValue} ${violationUnit}
+    Number of samples checked: ${numSamples}
+    
+    **Investigate more:** ${investigateUri}
+    `;
+      // [END v2PerformanceEventPayload]
+
+      try {
+        // [START v2SendToDiscord]
+        const response = await postMessageToDiscord(
+            "Firebase Performance Bot", message);
+        if (response.ok) {
```

Comment:
Do you think we should pull this code (probably also the `try`/`catch`) into `postMessageToDiscord()`? It's not replicated three times with only minimal differences in the info/error message being logged. We could just make the error messages more generic.

The complication is that this error handling is currently in the snippets (`[START v2SendToDiscord]`/`[END v2SendToDiscord]`)...

Reply:
I discussed this with Victoria, I think what we can do is leave this as is, but in the documentation, just link to the code snippet without showing the code itself, so it doesn't look as redundant in the doc. wdyt?

Ideally we can clean this up in the code snippets as well but as you said, it's currently used in the doc...

Reply:
Ok. We can always to the de-duplication in a follow-up.